jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -438,9 +439,14 @@ def _extract_from_aperture(self, cube, uncert_cube, aperture,
                 subset_id=aperture.selected, cls=NDDataArray
             )
             if uncert_cube:
-                uncertainties = uncert_cube.get_subset_object(
-                    subset_id=aperture.selected, cls=StdDevUncertainty
-                )
+                try:
+                    # Subset may not be linked to the uncertainty cube at this point??
+                    uncertainties = uncert_cube.get_subset_object(
+                        subset_id=aperture.selected, cls=StdDevUncertainty
+                    )
+                except ValueError:
+                    raise
```

Comment:
```suggestion
                    msg = SnackbarMessage(
                        f"Extracting uncertainties on subset for {aperture.selected} failed",  # noqa
                        color='error', sender=self, timeout=10000)
                    self.app.hub.broadcast(msg)
```
This would at least let the subset propagate to the spectral extraction tool, even if uncertainties are (initially) missing. I'd expect them to be available after the first modification (move/resize) to the subset, since I don't see the snackbar message repeated then, but could not find anything where `nddate.uncertainty` is used.